File: src/lib/preproc.py
def preprocessor(f):
    # remove comments
    workl = f
    for n, line in enumerate(f):
        if '#' not in line:
            pass
        else:
            workl[n] = workl[n][0:line.index('#')]
    for n in range(workl.count('')):
        del workl[workl.index('')]
    return workl

File: src/preproc.rs
// old Python file
// def preprocessor(stringlist):
//     # remove comments
//     workl = stringlist
//     linenum = 0
//     for n, line in enumerate(stringlist):
//         if '#' not in line:
//             workl[n] = (linenum, line)
//             linenum += 1
//         else:
//             workl[n] = (linenum, line[0:line.index('#')])
//             linenum += 1
//     return workl
